From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Felix Hanau <felix@cloudflare.com>
Date: Wed, 26 Jul 2023 18:40:13 +0200
Subject: Fix V8/ICU build

1) ICU changed how a header is included, but V8's bazel build file has not been
   updated to make the header available.
2) Bazel's V8 build will use abseil in the near future. While it is
   already added as a dependency, it is not actually required yet, so
   do not include it for now. In a future version, we can use our
   existing absl dependency, we'll just need to keep it in sync with
   V8's version.

diff --git a/BUILD.bazel b/BUILD.bazel
index a1823f5e7a88dc0417083f3d0c14003028aaa235..da3a7285ced73c3405da14e333c5e02020b2c280 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4272,7 +4272,6 @@ v8_binary(
         "UNISTR_FROM_CHAR_EXPLICIT=",
     ],
     deps = [
-        "//external:absl_optional",
         "v8_libbase",
         "//external:icu",
     ],
diff --git a/bazel/BUILD.icu b/bazel/BUILD.icu
index 5127ceb7b783b11d5750c01977d7e34606c39667..c81f792bf642e4ab42626dbf709c8c012433ddca 100644
--- a/bazel/BUILD.icu
+++ b/bazel/BUILD.icu
@@ -90,7 +90,7 @@ cc_library(
     hdrs = glob([
         "source/common/unicode/*.h",
         "source/i18n/unicode/*.h",
-    ]),
+    ]) + ["source/stubdata/stubdata.h"],
     copts = select({
         "@platforms//os:windows": [
             "/wd4005",  # Macro redefinition.
